design/ceph/ceph-nfs-ganesha.md

```diff
@@ -213,8 +225,115 @@ The following directories should **not** be shared between SSSD and other contai
 - `/run/dbus`: using the DBus instance from the sidecar caused SSSD errors in testing. SSSD only
   uses DBus for internal communications and creates its own socket as needed.
 
+### Kerberos
+Kerberos is the authentication mechanism natively supported by NFS-Ganesha.
+
+#### Kerberos principals
+The Kerberos service principal used by NFS-Ganesha to authenticate with the Kerberos server is built
+up from 3 components:
+1. a configured PrincipalName (from `spec.security.kerberos.principalName` in Rook) that acts as the
+   service name
+2. the hostname of the server on which NFS-Ganesha is running (using kernel call `getaddrinfo()`)
+3. the realm (as configured by the user's krb.conf file(s))
+
+The full service principal name is constructed as `<principalName>/<hostname>@<realm>`.
+
+Users must add this service principal to their Kerberos server configuration. Therefore, this
+principal must be static, and for the benefit of users it should be deterministic.
+
+The hostname of Kubernetes pods is the partly-random name of the pod by default.
+In order to give NFS-Ganesha server pods a deterministic hostname, the `hostname` field of the pod
```

Comment:
The `hostname` is set as an env var, or what field?

Reply:
Pods have a `hostname` field that can be set in the spec.